### ISSUE TITLE
Advanced spend burn more hours if we send to the same address

### ISSUE THREAD

**Issue by therealssj** [skycoin/skycoin#1419]
Example transaction:
```
{
    "transaction": {
      ...
        "fee": "3768",
        "sigs": [... ],
        "inputs": [
            {
                "uxid": "2991b8df8e8dcd55d5dd27a1748dbb2f812a93fd4deffc52c8de7a128d3f0a6b",
                "address": "tWPDM36ex9zLjJw1aPMfYTVPbYgkL2Xp9V",
                "coins": "13.000000",
                "calculated_hours": "7120",
            },
            {
                "uxid": "f5f838edf75b68882cacb7fa071538bcf800515d5a7f42e3a8c5e6d681879a82",
                "address": "tWPDM36ex9zLjJw1aPMfYTVPbYgkL2Xp9V",
                "coins": "0.500000",
                "calculated_hours": "416",
            }
        ],
        "outputs": [
            {
                "uxid": "a3d1cf25c4493a74085c16dd6a207bc606e4265d3385d5338357d340ec2ed676",
                "address": "tWPDM36ex9zLjJw1aPMfYTVPbYgkL2Xp9V",
                "coins": "13.000000",
                "hours": "1780"
            },
            {
                "uxid": "07d9da209d92b540c92fcde1c5be7348b23656ba391b0cda83943ada3c9d4036",
                "address": "tWPDM36ex9zLjJw1aPMfYTVPbYgkL2Xp9V",
                "coins": "0.500000",
                "hours": "1988"
            }
        ]
    },
}
```
in the above transaction we can just send the hours to the receiving address since its the same address and burn less hours `(7120/2)` instead of `((7120+416)/2)` like in the case above

**Comment by gz-c** [comment]
7120-1780 = 5340 hours lost

**Comment by gz-c** [comment]
No extra change, respect 50% share factor
- 7120 hours lost by spender
- 1780 hours to receiver
- 5340 hours burned

**Result:** Maximum hours lost by spender, maximum hours burned

No extra change, switch to 100% share factor 
- 7120 hours lost by spender
- 3560 hours to receiver
- 3560 hours burned

**Result:** Maximum hours lost by spender, minimum hours burned

Extra change, respect 50% share factor based upon initial chosen outputs
- 7120+416-1988 = 5548 hours lost by spender
- 1780 hours to receiver
- 3768 hours burned

**Result:** Minimum hours lost by spender, slightly more hours burned

Extra change, respect 50% share factor based upon final chosen outputs [too complicated to implement and not what is intended]
- 7120+416-1884 = 5652 hours lost by spender
- 1884 hours to receiver
- 3768 hours burned

**Result:** Slightly more hours lost by spender, slightly more hours burned


When the sender is the same person as the receiver, they only care about minimizing
hours burned. In this case, switching to 100% share factor is ideal, which is
what /wallet/spend does. Otherwise, the extra change mechanism is ideal for 
preserving the sender's coin hours.

---------

As a fallback, if extra change cannot be created to preserve coin hours, we could
redistribute the leftover hours to the destinations.  This would help for the
case of sending all of your coins from one wallet to another 
(since you're sending all, you don't have any more coins to make change with).

However, this behavior could confuse the user.  
This can only be fully solved with UI changes, see below.

----------

Redistributing the extra coin hours instead of creating an extra change output
is not always the best decision.  For manually specified coin hours, we should send
exactly the hours specified, so we couldn't do it in that case. 

For an example of an edge case where extra change output is sensible,
supposed you had two outputs:

<coins=1, hours=100000>, <coins=10000, hours=0>

and you wanted to send 1 coin to another person.  If you did not do the extra change method,
you would lose all 100000 coin hours.

--------

For the general case, the UI will need to give the user control over the share factor.

Some ideas:

- A toggle/switch between "share hours" (uses 0.5 sf), "send all hours" (uses 1.0 sf)
- A slider over a range of 0.5-1.0 

Basically we want the default to be 0.5 share factor, but if they want to send to themselves
they probably want to do 1.0 share factor.
We don't want to make it easy for them to do 0.0 share factor because we want coin hours spread around.
If you received 0.001 coins and no hours you wouldn't be able to move it for 42 days.
Of course this can still happen with 0.5 sf if the input coin hours are low enough, but we want to minimize it.

**Comment by gz-c** [comment]
Issues created from the above:
https://github.com/skycoin/skycoin/issues/1421 
https://github.com/skycoin/skycoin/issues/1422
